### DISCUSSION TITLE
Please consult and ask a logic application question，Client Change Public IP address

### DISCUSSION THREAD

**Discussion by TelDragon** [strongswan/strongswan#1976] [Q&A] [answered]
Company A Network Environment

The network public network Public IP address address is as shown in 101.101.101.101.101
The internal private IP address range is as shown in 172.16.1.1-255
The IP address of a certain service physical machine is 172.16.1.2

This environment uses the strongswan Server mode, create Authentication Type do IKEv2 EAP (Username/Password)
pools Addresses* 10.10.10.0/24

```
iptables -t nat -A POSTROUTING -s 10.10.10.0/24 -o eth0 -j MASQUERADE
```

Company B Network Environment

The network public network Public IP address address is as shown in 202.202.202.202
The internal private IP address range is as shown in 192.168.1.1~255
The IP address of a certain service physical machine is 192.168.1.5

This environment uses the strongswan Client mode, connect strongswan Server 101.101.101.101.101
Visit 172.16.1.2 resources in Company A, or other resources

After connecting to the strongswan Server normally on the client used by Company B. 
Discovered that the Public IP address of this client has changed from the original 202.202.202.202 to 101.101.101.101.101

What should I do if I don't want to change the Public IP address  of Company B client?

**Comment by tobiasbrunner** [comment]
What do you mean with "export IP"? It's public IP? Sounds like you are tunneling everything to the other peer. If you don't want that, change the traffic selectors.

**Reply by TelDragon** [reply]
Sorry, respected author, my translation may not be very accurate.

It is indeed an Public IP address.

Are the traffic selectors you mentioned "Local traffic selectors" in strongMan Server Mode? Can I fill it out this way? 172.16.1.0/24

**Reply by tobiasbrunner** [reply]
I don't know what an export IP is, sorry. But yes, strongMan does have a setting to change the local traffic selector.

**Reply by TelDragon** [reply]
I'm very sorry!, It should be understood in this way.    Public IP address  、Internet address 。

Are you sure you want to change the Local traffic selector project or the Remote traffic selector???

**Reply by tobiasbrunner** [reply]
On the server, the local traffic selectors are the ones the clients will have access to, the remote traffic selector will be the virtual IP address, so don't specify it in the GUI.

**Reply by TelDragon** [reply]
Oh. Okay, I'm sorry for confusion.

So in my situation, I don't want to change the client's public IP address or internet IP address. What should I do?

**Reply by tobiasbrunner** [reply]
I thought you wanted to avoid that everything is tunneled from site B to site A, i.e. only traffic to `172.16.1.0/24` should go through the tunnel, no? If that's the case, configure that subnet as local traffic selector in the server config (instead of `0.0.0.0/0`, which you presumably configured now).

**Comment by TelDragon** [comment]
> I thought you wanted to avoid that everything is tunneled from site B to site A, i.e. only traffic to 172.16.1.0/24 should go through the tunnel, no? If that's the case, configure that subnet as local traffic selector in the server config (instead of 0.0.0.0/0, which you presumably configured now).


Sorry, when I changed the Local traffic selector to 172.16.1.0/24. Clients in Company B can access 172.16.1.2.
But client B cannot access the internet anymore. Do I need to operate additional iptables rules?
Browser cannot access the internet

**Reply by tobiasbrunner** [reply]
I guess this depends on the client (e.g. if it blocks traffic that bypasses the VPN, in our Android app you have to allow that explicitly). Also see the [notes on split-tunneling](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_split_tunneling).

**Reply by TelDragon** [reply, marked as answer]
My current server's iptables firewall rule settings have been added
```
iptables -t nat -A POSTROUTING -s 10.10.10.0/24 -d 172.16.1.0/24  -o eth0 -j MASQUERADE
```

Display of packages
```
iptables -vnL -t nat
Chain POSTROUTING (policy ACCEPT 8112 packets, 573K bytes)
 pkts bytes target     prot opt in     out     source               destination
    8   480 MASQUERADE  all  --  *      *       10.10.10.0/24        172.16.1.0/24  
```

strongMan Server Traffic selectors
```
Local traffic selector          172.16.1.0/24  
Remote traffic selector
```

The current state is that after the Windows 10 client of Company B is connected, it can access the different IP addresses of the 172.16.1.2 and 172.16.1.0 network segments normally.
```
tracert 172.16.1.3

Tracking through up to 30 hops
Route to Bogon [172.16.1.3]:

  1    18 ms    18 ms    18 ms  bogon [172.16.1.2]
  2    22 ms    18 ms    18 ms  bogon [172.16.1.3]

Tracking completed.
```

After logging in to the strongSwan mobile app, his public IP remains unchanged at 202.202.202.202. And can access addresses 172.16.1.2-255 normally


After using strongMan as the client on my debian11 and connecting to the strongMan EAP Server normally. Its public IP has not changed either. And can access the internal network assets in Company A normally 172.16.1.2-255


Now it's basically clear. It may be a problem with the Windows 10 client.，What a mess~
For my Windows 10 client, after checking, I found that it does indeed have a split tunnel enabled by default. SplitTunneling
```
PS C:\Windows\system32> Get-VpnConnection


Name                  : ipsec.xxxx.com.cn
ServerAddress         : ipsec.xxxx.com.cn
AllUserConnection     : False
Guid                  : {AF34E98E-939D-462A-9BAA-D5A69BDA7F5F}
TunnelType            : Ikev2
AuthenticationMethod  : {Eap}
EncryptionLevel       : Optional
L2tpIPsecAuth         :
UseWinlogonCredential : False
EapConfigXmlStream    : #document
ConnectionStatus      : Connected
RememberCredential    : True
SplitTunneling        : True
DnsSuffix             :
IdleDisconnectSeconds : 0
```

After continuous testing, I forgot the order. It may be necessary to join the VPN route in advance, and it works normally

Ah。 Oh my goodness. I am very depressed. The operation of adding a VPN route must be done before connecting to the VPN
```
Add-VpnConnectionRoute -ConnectionName "ipsec.xxxx.com.cn" -DestinationPrefix "172.16.1.0/24" -PassThru

DestinationPrefix : 172.16.1.0/24
InterfaceIndex    :
InterfaceAlias    : ipsec.xxxx.com.cn
AddressFamily     : IPv4
NextHop           : 0.0.0.0
Publish           : 0
RouteMetric       : 1
PolicyStore       :
```
